docs/advanced-usage/miscellaneous.md

```diff
@@ -17,6 +17,53 @@ void loop() {
 }
 ```
 
+# Organizing the sketch in good and safe order
+
+The example above is good to demonstrate usage of `Homie.isConnected()`. However, consider do not adding any code into `loop()` function. ESP8266 is relatively weak MCU, and Homie manages few important environmental tasks. Interferring with Homie might boomerang by sudden crashes. Advised sketch of Homie is:
+```c++
+#include <Homie.h>
+
+void setupHandler() {
+  // Code which should run AFTER the WiFi is connected.
+}
+
+void loopHandler() {
+  // Code which should run in normal loop(), after setupHandler() finished.
+}
+
+void setup() {
+  Serial.begin(115200);
+  Serial << endl << endl;
+
+  Homie_setFirmware("bare-minimum", "1.0.0"); // The underscore is not a typo! See Magic bytes
+  Homie.setSetupFunction(setupHandler).setLoopFunction(loopHandler);
+  
+ // Code which should run BEFORE the WiFi is connected.
+
+  Homie.setup();
+}
+
+void loop() {
+  Homie.loop();
+}
+```
+
+This way you can be sure the run is safe enough (unless you use blocking delay, then it is not) and the sketch keeps 'regular' structure of `setup()`/`loop()` just in terms of Homie.
+
+## Why it's needed `setupHandler()` in addition to `setup()` in Homie?
+`setup()` starts execution immediately after power on/wake up. `setupHandler()` starts to run when WiFi established the connection.
+
+## Why it's needed `loopHandler()` in addition to `loop()` in Homie?
+`loop()` starts to run then `setup() is completed`. This somehow controls a run of `setupHandler()`. The `loopHandler()` serves safe way to run the loop.
```

Comment:
- to run when `setup()` is completed
- again `setupHandler()`?

Reply:
- Maybe "after" suits better than "when" or "then"?
- Yes. From my understanding, the Arduino loop() calls Homie.loop(). Homie detects what's already done, what's the statusl and first thing launches "setupHandler" and then switches to "loopHandler". Not sure thus I understood this 100% correctly.